pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,170 @@
+	shared.LogLevel("INFO", "TEST: 'CLASSIC' Secrets Encryption method")
+	index := len(nodes) - 1
+	secretsEncryptOps("prepare", product, nodes[index].ExternalIP, ips)
+	secretsEncryptOps("rotate", product, nodes[index].ExternalIP, ips)
+	secretsEncryptOps("reencrypt", product, nodes[index].ExternalIP, ips)
+
+	if strings.Contains(os.Getenv("TEST_TYPE"), "both") {
+		shared.LogLevel("INFO", "TEST: 'NEW' Secrets Encryption method")
+		secretsEncryptOps("rotate-keys", product, nodes[index].ExternalIP, ips)
+	}
+}
+
+func secretsEncryptOps(action, product, cpIp string, ips []string) {
+	shared.LogLevel("INFO", fmt.Sprintf("TEST: Secrets-Encryption: %s", action))
+	_, errStatusB4 := shared.SecretEncryptOps("status", cpIp, product)
+	Expect(errStatusB4).NotTo(HaveOccurred(), "error getting secret-encryption status before action")
+
+	stdOutput, err := shared.SecretEncryptOps(action, cpIp, product)
+	Expect(err).NotTo(HaveOccurred(), "error: secret-encryption: "+action)
+	verifyStdOut(action, stdOutput)
+	if (action == "reencrypt") || (action == "rotate-keys") {
+		shared.LogLevel("DEBUG", "reencrypt op needs some time to complete - Sleep for 20 seconds before service restarts")
+		time.Sleep(20 * time.Second) // Wait for reencrypt action to complete before restarting services
+	}
+	for i, node := range ips {
+		nodearr := []string{node}
+		nodeIp, errRestart := shared.ManageService(product, "restart", "server", nodearr)
+		Expect(errRestart).NotTo(HaveOccurred(), "error restart service for node: "+nodeIp)
+		// Order of reboot matters. Etcd first then control plane nodes.
+		// Little lag needed between node restarts to avoid issues.
+		waitEtcdErr := shared.WaitForPodsRunning(5, 4, false)
+		if waitEtcdErr != nil {
+			shared.LogLevel("WARN", "pods not up after 20 seconds.")
+			if i != len(ips) {
+				shared.LogLevel("DEBUG", "continue service restarts")
+			}
+		}
+	}
+	switch product {
+	case "k3s":
+		waitPodsErr := shared.WaitForPodsRunning(5, 6, false)
+		if waitPodsErr != nil {
+			shared.LogLevel("WARN", "pods not up after 30 seconds")
+		}
+	case "rke2":
+		waitPodsErr := shared.WaitForPodsRunning(5, 12, false)
+		if waitPodsErr != nil {
+			shared.LogLevel("WARN", "pods not up after 60 seconds")
+		}
+	}
+
+	secretEncryptStatus, errGetStatus := waitForHashMatch(cpIp, product, 5, 36) // Max 3 minute wait time for hash to match
+	Expect(errGetStatus).NotTo(HaveOccurred(), "error getting secret-encryption status")
+	verifyStatusOutput(action, secretEncryptStatus)
+
+	errLog := logEncryptionFileContents(ips, product)
+	Expect(errLog).NotTo(HaveOccurred())
+}
+
+func waitForHashMatch(cpIp, product string, defaultTime time.Duration, times int) (string, error) {
+	var secretEncryptStatus string
+	var errGetStatus error
+	for i := 1; i <= times; i++ {
+		secretEncryptStatus, errGetStatus = shared.SecretEncryptOps("status", cpIp, product)
+		if errGetStatus != nil {
+			shared.LogLevel("DEBUG", "error getting secret-encryption status. Retry.")
```

Comment:
this is not a WARN ?

Reply:
no. the first few times, we wont get the secret-encryption status. it will be a few tries before the results start coming.

Reply:
ok tks